From cf8a2f9d9e938a580027e02d3418a0fed62c6b36 Mon Sep 17 00:00:00 2001
From: hauno <imhauno@gmail.com>
Date: Tue, 6 May 2014 18:01:37 -0700
Subject: [PATCH] Cap Objective Score Length

Adds a check for Score arguments that would crash the client

diff --git a/src/main/java/org/bukkit/craftbukkit/scoreboard/CraftObjective.java b/src/main/java/org/bukkit/craftbukkit/scoreboard/CraftObjective.java
index 9eaec71..0ee1147 100644
--- a/src/main/java/org/bukkit/craftbukkit/scoreboard/CraftObjective.java
+++ b/src/main/java/org/bukkit/craftbukkit/scoreboard/CraftObjective.java
@@ -95,6 +95,7 @@ final class CraftObjective extends CraftScoreboardComponent implements Objective
 
     public Score getScore(String entry) throws IllegalArgumentException, IllegalStateException {
         Validate.notNull(entry, "Entry cannot be null");
+        if (entry.length() > 16) throw new IllegalArgumentException("Entry cannot be longer than 16 characters!"); // Spigot
         CraftScoreboard scoreboard = checkState();
 
         return new CraftScore(this, entry);
-- 
1.9.1

